src/pages/registration/resource_type_tab/components/RelatedResultsField.tsx

```diff
@@ -0,0 +1,114 @@
+import { ResourceFieldNames } from '../../../../types/publicationFieldNames';
+import {
+  ConfirmedDocument,
+  createEmptyUnconfirmedDocument,
+  UnconfirmedDocument,
+} from '../../../../types/registration.types';
+import { dataTestId } from '../../../../utils/dataTestIds';
+import { RelatedResultItem } from '../sub_type_forms/research_data_types/RelatedResultItem';
+import { SearchRelatedResultField } from './SearchRelatedResultField';
+
+export const RelatedResultsField = () => {
+  const { t } = useTranslation();
+  const { values, setFieldValue } = useFormikContext<DegreeRegistration>();
+
+  const related = values.entityDescription.reference?.publicationInstance.related ?? [];
+
+  const removeRelatedItem = (indexToRemove: number) => {
+    const newRelated = related?.filter((_, thisIndex) => thisIndex !== indexToRemove);
+    setFieldValue(ResourceFieldNames.PublicationInstanceRelated, newRelated);
+  };
+
+  const moveRelatedResult = (newSequence?: number, oldSequence?: number) => {
+    if (!newSequence || !oldSequence) return;
+
+    const oldIndex = related.findIndex((document) => document.sequence === oldSequence);
+    const minNewIndex = 0;
+    const maxNewIndex = related ? related.length : 0;
+
+    const newIndex =
+      newSequence - 1 > maxNewIndex
+        ? maxNewIndex
+        : newSequence < minNewIndex
+          ? minNewIndex
+          : related
+            ? related.findIndex((document) => document.sequence === newSequence)
+            : 0;
+
+    const orderedRelatedResults =
+      newIndex >= 0 && related
+        ? (move(related, oldIndex, newIndex) as (ConfirmedDocument | UnconfirmedDocument)[])
+        : related;
+
+    // Ensure incrementing sequence values
+    const newRelatedResults = orderedRelatedResults
+      ? orderedRelatedResults.map((related, index) => ({
+          ...related,
+          sequence: index + 1,
+        }))
+      : [];
+
+    setFieldValue(ResourceFieldNames.PublicationInstanceRelated, newRelatedResults);
+  };
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        gap: '0.5rem',
+        bgcolor: 'secondary.light',
+        borderRadius: '0.25rem',
+        p: '1rem',
+      }}>
+      <Typography variant="h2">{t('registration.resource_type.related_results')}</Typography>
+
+      <SearchRelatedResultField />
+
+      <Trans
+        i18nKey={'registration.resource_type.related_results_description'}
+        components={[
+          <Typography key={0}>
+            <span style={{ fontWeight: 'bold' }} />
+          </Typography>,
+        ]}
+      />
+      <FieldArray name={ResourceFieldNames.PublicationInstanceRelated}>
+        {({ push }: FieldArrayRenderProps) => (
+          <Button
+            variant="outlined"
+            data-testid={dataTestId.registrationWizard.resourceType.addRelatedButton}
+            onClick={() => push(createEmptyUnconfirmedDocument(related.length + 1))}
```

Comment:
personlig liker jeg nok best når dette er eksplisitt, da jeg ikke kan vite at arumentet der settes til sequence uten å sjekke `createEmptyUnconfirmedDocument`. Men det er vel en ren subjektiv sak, så kan godt være sånn også :)
```suggestion
            onClick={() => push({
               ...emptyUnconfirmedDocument, 
               sequence: related.length + 1,
             })}
```

Reply:
Enig der👍